app/controllers/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UsersController < ApplicationController
+
+  get '/' do
+    erb :index
+  end
+
+  get "/users/:slug" do
+    @user = User.find_by_slug(params[:slug])
+    session[:user_id] = @user.id
+    erb :'/tweets'
+  end
+
+  get '/signup' do
+    if !Helpers.is_logged_in?(session)
+      erb :'users/create_user'
+    else
+      redirect to '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if !params["username"].empty? && !params["email"].empty? && !params["password"].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.